packages/core/src/tx.ts

```diff
@@ -117,6 +116,30 @@ export function withOperations<T extends WithTx> (user: Ref<Account>, storage: T
     return TxProcessor.createDoc2Doc(tx) as T
   }
 
+  result.createDocWithShortRef = async <T extends Doc>(
+    _class: Ref<Class<T>>,
+    space: Ref<Space>,
+    attributes: Data<T>
+  ): Promise<T> => {
+    const tx: TxCreateDoc<T> = {
+      _id: generateId(),
+      _class: core.class.TxCreateDoc,
+      space: core.space.Tx,
+      modifiedBy: user,
+      modifiedOn: Date.now(),
+      objectId: generateId(),
+      objectClass: _class,
+      objectSpace: space,
+      attributes
+    }
+    await storage.tx(tx)
+    const result = TxProcessor.createDoc2Doc(tx) as T
+    const objectSpace = (await storage.findAll(core.class.Space, { _id: result.space }, { limit: 1 }))[0]
+    const workspace = objectSpace.name.trim().toUpperCase().replace(/[^a-z0-9]/gim, '_').replace(/[_]/g, '_')
+    await createShortRef(storage, user, result, workspace)
```

Comment:
For task I've propose to store short Id as part of object itself as well to not perform extra find requests, since it will be good to see it in most places task will be shown.